### PR TITLE
Embench deployment to Core-v-verif for cv32e40p

### PR DESCRIPTION
**Pull request by silabs-mateilga**
Signedoff-by: Marton Teilgard <mateilga@silabs.com>
This is the deployment of EMBench support for Core-v-verif, as described in issue #495 .

The commit includes:

- EMBench integration via make environment
- The ability to build and run benchmarks for speed and size
- Option for setting target score that pass/fails the benchmark run
- Test artifacts than can be simulated through "make test"

Details can be found in the readme file:
core-v-verif/cv32e40p/tests/embench/README.md

### CODE REVIEW

**Comment by MikeOpenHWGroup** [comment]
Hi @silabs-mateilga, you will need to signoff your git commit before this pull-request can be merged in.   I _believe_ you can do this with the "amend" command-line argument:
```
$ git commit --amend -s -m 'Adding signoff'
$ git push silabs-meteilga embench_integration_1
```
Note that I say "believe", because I have not done this myself.  :stuck_out_tongue_winking_eye:

File: bin/templates/test.yaml.j2
Comment (MikeOpenHWGroup) [inline bin/templates/test.yaml.j2:1]: Can we have a more descriptive name for this file?   Ideally it should be obvious from the filename that this is running the embench tests.

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
Hi @silabs-mateilga.  Impressive work!   I've made a few comments.   I will also be marking this PR as "do not merge" until I can determine whether the files covered by the GPL are safe for us to merge.

File: bin/templates/test.yaml.j2
Comment (silabs-mateilga) [reply bin/templates/test.yaml.j2:1]: Yes, this is possible. The reason it's originally named "test.yaml.j2" is that it matches the name of the file(s) that are generated from this template, "test.yaml". If you prefer the naming to match the use, say "embench_test.yaml.j2", that is an easy fix.

File: cv32e40p/tests/.gitignore
Comment (silabs-mateilga) [reply cv32e40p/tests/.gitignore:17]: Yes we can! Good catch, helps with future proofing

File: mk/Common.mk
Comment (silabs-mateilga) [reply mk/Common.mk:70]: At this point I'm on not sure if I can make a good call, as I've copied the method from the existing make method for the other external repos, like RISCVDV. Maybe @strichmo can comment on this?

File: cv32e40p/tests/embench/config/corev32/boards/corev32/board.cfg
Comment (silabs-mateilga) [reply cv32e40p/tests/embench/config/corev32/boards/corev32/board.cfg:1]: Oversight, will be fixed in next commit

File: mk/Common.mk
Comment (MikeOpenHWGroup) [reply mk/Common.mk:70]: Yeah, this sort of got away on us.   The question is, what are the minimum set of shell env variables that must be set?  Obviously RISCV or COREV must be set for all simulations.  Beyond that it is not clear.  We could make a good case for RISCVDV since most users will be using the ISG.  Beyond that, things like the Compliance suite and Embench suite may or may not be required.

**Comment by silabs-mateilga** [comment]
Fixed the simple issues rasied by @MikeOpenHWGroup , awaiting resolution on remaining discussion.

File: mk/Common.mk
Comment (strichmo) [reply mk/Common.mk:70]: Let's make this a warning.

Comment (silabs-mateilga) [reply mk/Common.mk:70]: Changed to warning

**Review by MikeOpenHWGroup — APPROVED** [review]
Looks good @silabs-mateilga, thanks for the updates.   I'm running a ci_check locally and will merge this in when complete.

**Pull request merged**
